charts/latest/azuredisk-csi-driver/templates/csi-azuredisk-node-windows.yaml

```diff
@@ -33,16 +33,52 @@ spec:
 {{ toYaml .Values.windows.nodeAffinity | indent 10 }}
       priorityClassName: system-node-critical
       {{- include "azuredisk.pullSecrets" . | indent 6 }}
+{{- if eq .Values.windows.useHostProcesContainers true }}
+      securityContext:
+        windowsOptions:
+          hostProcess: true
+          runAsUserName: "NT AUTHORITY\\SYSTEM"
+      hostNetwork: true
+{{- end }}
+{{- if eq .Values.windows.useHostProcessContainers true }}
+      initContainers:
+        - name: init
+{{- if hasPrefix "/" .Values.image.azuredisk.repository }}
+          image: "{{ .Values.image.baseRepo }}{{ .Values.image.azuredisk.repository }}:{{ .Values.image.azuredisk.tag }}"
+{{- else }}
+          image: "{{ .Values.image.azuredisk.repository }}:{{ .Values.image.azuredisk.tag }}"
+{{- end }}
+          imagePullPolicy: {{ .Values.image.azuredisk.pullPolicy }}
+          command:
+            - "powershell.exe"
+            - "-c"
+            - "New-Item"
+            - "-ItemType"
+            - "Directory"
+            - "-Path"
+            - "C:\\var\\lib\\kubelet\\plugins\\{{ .Values.driver.name }}\\"
+            - "-Force"
+{{- end }}
       containers:
         - name: liveness-probe
-          volumeMounts:
-            - mountPath: C:\csi
-              name: plugin-dir
 {{- if hasPrefix "/" .Values.image.livenessProbe.repository }}
           image: "{{ .Values.image.baseRepo }}{{ .Values.image.livenessProbe.repository }}:{{ .Values.image.livenessProbe.tag }}"
 {{- else }}
           image: "{{ .Values.image.livenessProbe.repository }}:{{ .Values.image.livenessProbe.tag }}"
 {{- end }}
+          imagePullPolicy: {{ .Values.image.livenessProbe.pullPolicy }}
```

Comment:
I thought it was a small diff, while it looks like these two deployments are quite different now, then I think we should keep two deployment files, otherwise it may be hard to understand and maintain.

Reply:
No problem. I dropped the commit to split merge the manifests.
Let me rebase again.
The manifest do have many subtle differences.